event/register_oracle_event.go

```diff
@@ -49,11 +58,90 @@ func (e RegisterOracleEvent) EventHandler(event ctypes.ResultEvent) error {
 
 	nodePubKeyHash := sha256.Sum256(oracleRegistration.NodePubKey)
 
-	err = sgx.VerifyRemoteReport(oracleRegistration.NodePubKeyRemoteReport, nodePubKeyHash[:], *e.reactor.EnclaveInfo())
+	voteOption := verifyReportAndGetVoteOption(oracleRegistration, nodePubKeyHash, e)
+
+	msgVoteOracleRegistration, err := makeOracleRegistrationVote(uniqueID, e.reactor.OracleAcc().GetAddress(), addressValue, voteOption, e.reactor.OraclePrivKey())
+	if err != nil {
+		return err
+	}
+
+	txBytes, err := generateTxBytes(msgVoteOracleRegistration, e.reactor.OracleAcc().GetPrivKey(), e.reactor.Config(), e.reactor.TxBuilder())
 	if err != nil {
 		return err
 	}
 
-	// TODO: Executing Vote Txs
+	if err := broadCastTx(e.reactor.GRPCClient(), txBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// verifyReportAndGetVoteOption validates the RemoteReport and returns the voting result according to the verification result.
+func verifyReportAndGetVoteOption(oracleRegistration *types.OracleRegistration, nodePubKeyHash [32]byte, e RegisterOracleEvent) types.VoteOption {
+	if err := sgx.VerifyRemoteReport(oracleRegistration.NodePubKeyRemoteReport, nodePubKeyHash[:], *e.reactor.EnclaveInfo()); err != nil {
+		log.Infof("failed to verification report. uniqueID(%s), address(%s), err(%v)", oracleRegistration.UniqueId, oracleRegistration.Address, err)
+		return types.VOTE_OPTION_NO
+	} else {
+		return types.VOTE_OPTION_YES
+	}
+}
```

Comment:
When I think about it again, changing it like this makes it clearer. :)

```suggestion
func verifyReportAndGetVoteOption(oracleRegistration *types.OracleRegistration, enclaveInfo sgx.EnclaveInfo) types.VoteOption {
	nodePubKeyHash := sha256.Sum256(oracleRegistration.NodePubKey)
	if err := sgx.VerifyRemoteReport(oracleRegistration.NodePubKeyRemoteReport, nodePubKeyHash[:], enclaveInfo); err != nil {
		log.Infof("failed to verification report. uniqueID(%s), address(%s), err(%v)", oracleRegistration.UniqueId, oracleRegistration.Address, err)
		return types.VOTE_OPTION_NO
	} else {
		return types.VOTE_OPTION_YES
	}
}
```

Reply:
Oh, yes. I'll change it.